index.js

```diff
@@ -0,0 +1,213 @@
+}
+
+function setActive(eventCode) {
+  const elem = document.getElementById(eventCode);
+  if (!elem.parentElement.className.includes(' active')) elem.parentElement.className += ' active';
+}
+
+function printText(eventCode) {
+  const elem = document.getElementById(eventCode);
+  const textArea = document.querySelector('.textArea');
+  textArea.textContent += elem.textContent;
+}
+
+function changeLang(eventCode) {
+  State[eventCode] = true;
+  if (State.ControlLeft && State.AltLeft) {
+    switch (localStorage.lang) {
+      case 'ruDown': {
+        initKeyboard(enDown);
+        localStorage.lang = 'enDown';
+        break;
+      }
+      case 'enDown': {
+        initKeyboard(ruDown);
+        localStorage.lang = 'ruDown';
+        break;
+      }
+      case 'ruUp': {
+        initKeyboard(enUp);
+        localStorage.lang = 'enUp';
+        break;
+      }
+      case 'enUp': {
+        initKeyboard(ruUp);
+        localStorage.lang = 'ruUp';
+        break;
+      }
+      default: break;
+    }
+  }
+}
+
+function capslock(shiftFlag) {
+  if (State.capslock && !shiftFlag) {
+    return;
+  }
+  State.capslock = true;
+  switch (localStorage.lang) {
+    case 'ruDown': {
+      initKeyboard(ruUp);
+      localStorage.lang = 'ruUp';
+      break;
+    }
+    case 'enDown': {
+      initKeyboard(enUp);
+      localStorage.lang = 'enUp';
+      break;
+    }
+    case 'ruUp': {
+      initKeyboard(ruDown);
+      localStorage.lang = 'ruDown';
+      break;
+    }
+    case 'enUp': {
+      initKeyboard(enDown);
+      localStorage.lang = 'enDown';
+      break;
+    }
+    default: break;
+  }
+}
+
+function shift() {
+  if (!State.shift) {
+    capslock(true);
+  }
+  State.shift = true;
+}
+
+function keyDown(eventCode) {
+  setActive(eventCode);
+  const textArea = document.querySelector('.textArea');
+
+  switch (eventCode) {
+    case 'AltLeft':
+    case 'ControlLeft': changeLang(eventCode); return;
+    case 'ShiftRight':
+    case 'ShiftLeft': shift(); return;
+    case 'Backspace': textArea.textContent = textArea.textContent.slice(0, -1); return;
+    case 'Enter': textArea.textContent += '\n'; return;
+    case 'Tab': textArea.textContent += '   '; return;
+    case 'Delete': textArea.textContent = ''; return;
+    case 'CapsLock': capslock(); return;
+    case 'ControlRight': return;
+    case 'MetaLeft':
+    case 'AltRight': return;
+    default: break;
+  }
+
+  printText(eventCode);
+}
+
+function keyUp(eventCode) {
+  if (eventCode === 'ControlLeft' || eventCode === 'AltLeft') State[eventCode] = false;
+  if (eventCode.includes('Shift') && State.shift) {
+    capslock(true);
+    State.shift = false;
+  }
+  if (eventCode === 'CapsLock') {
+    State.capslock = false;
+  }
+
+  const elem = document.getElementById(eventCode);
+  elem.parentElement.className = elem.parentElement.className.replace(' active', '');
+}
+
+function listen() {
+  const elements = document.querySelectorAll('span');
+  elements.forEach((elem) => {
```

Comment:
there are a lot of listeners, it would be better to delegate events from keyboard container